src/components/alarm/AlarmController.cpp

```diff
@@ -88,21 +103,66 @@ uint32_t AlarmController::SecondsToAlarm() {
 
 void AlarmController::DisableAlarm() {
   xTimerStop(alarmTimer, 0);
-  state = AlarmState::Not_Set;
+  isAlerting = false;
+  if (alarm.isEnabled) {
+    alarm.isEnabled = false;
+    SaveAlarmToFile();
+  }
 }
 
 void AlarmController::SetOffAlarmNow() {
-  state = AlarmState::Alerting;
+  isAlerting = true;
   systemTask->PushMessage(System::Messages::SetOffAlarm);
 }
 
 void AlarmController::StopAlerting() {
-  // Alarm state is off unless this is a recurring alarm
-  if (recurrence == RecurType::None) {
-    state = AlarmState::Not_Set;
+  isAlerting = false;
+  // Disable alarm unless it is recurring
+  if (alarm.recurrence == RecurType::None) {
+    alarm.isEnabled = false;
+    SaveAlarmToFile();
```

Comment:
instead of saving the file on every little change I recommend doing it like the settings file. It has a `bool settingsChanged = false;` flag. This flag can be set on modification and only if changed write out the new alarm-settings file

Reply:
I think everywhere where you call `SaveAlarmToFile` you could instead set `settingsChanges = true;` and in the destructor of the Alarm call `SaveAlarmToFile` if `settingsChanges == true`

Am I missing something?

Reply:
I actually had that first, but the destructor does not get called if you don't close the alarm app, so we fail the user in the following scenario:
1. User opens the alarm app to set an alarm
2. User does not close the alarm app, watch turns off via idle timeout
3. Watch resets for some reason (maybe User was testing some fancy unstable feature)
The watch will load the last saved alarm when rebooting, either waking the user up at an unwanted time, or worse, not wake the user at the alarm set before the watch reset.

Reply:
another idea: let's save on exiting, and additionally immediately always when the on/off state is toggled. Then the enabled or disabled alarm is always stored. And loosing a disabled time because of not exiting the alarm app is tolerable

Reply:
See [`ce296ec` (#1333)](https://github.com/InfiniTimeOrg/InfiniTime/pull/1333/commits/ce296ece8335b31fe014d9ec0098689647172fea)

Reply:
One remaining inefficiency is that enabling saves twice: once for `SetAlarmTime` and again for `ScheduleAlarm`, but I think we can live with that. Avoiding that would cost more in binary size than it's worth IMHO.